### ISSUE TITLE
Additional features

### ISSUE THREAD

**Issue by ash123456789** [ninetynine/git-admin#1]
The following features could be extremely handy.

**Listing existing repositories**
Returns json of the existing repositories that the access token can see. This could also be expanded to specify a particular organisation.

Example:
```
git-admin repo list
```

With organisation:
```
git-admin repo list --org ninetynine
```

**Description & URL for creating and updating**
Updating the description and URL for an individual repository.

Example:
```
git-admin repo edit user/awesome-new-project --name radical-new-project --desc This is a description --url https://google.co.uk --private
```

**Default branch and branch protection rules**
It would be very nice if we could manage the default branch and add/remove branch protection rules.

Default Branch Example:
```
git-admin repo edit user/awesome-new-project --default-branch production
```

Protection Rules Example:
```
git-admin repo edit user/awesome-new-project --branch production --pr-reviews --status-checks --signed-commits --restrict-admins --restrict-push username-or-email
```

* `--branch` - Specifies the branch to apply the rules to.
* `--pr-reviews` - Require pull request reviews before merging.
* `--status-checks` - Require status checks to pass before merging.
* `--signed-commits` - Require signed commits.
* `--restrict-admins` - Include administrators (Enforce all configured restrictions for administrators).
* `--restrict-push` - Restrict who can push to matching branches.

**Comment by dextermb** [comment]
I believe that in the GitHub API the `--url` flag is called ["homepage"][homepage]. Would you still prefer "url"?

When setting protection for a repository what about:
```
git-admin repo protect user/awesome-new-project master [--pr-reviews] [-status-checks] [--signed-commits] [--restrict-admins] [--restrict-push]
```

[homepage]: https://developer.github.com/v3/repos/#input-1

**Comment by ash123456789** [comment]
The example commands are just very rough examples. As long as they are self-explanatory the name doesn't really matter. Personally, `homepage` or `url` is fine but I think `homepage` would make more sense.

As for the protection, the example you provided seems to be better.

**Comment by dextermb** [comment]
> Personally, `homepage` or `url` is fine but I think `homepage` would make more sense.

I think I will add aliases so that either can be done 👍

**Comment by dextermb** [comment]
More editable properties implemented in e35594483a5bd6e807563b9bbc9304a26c50f8c4.

Edit: Installable as of `^1.1.0-beta`

**Comment by dextermb** [comment]
> **Listing existing repositories**
> Returns json of the existing repositories that the access token can see. This could also be expanded to specify a particular organisation.

What kind of output are you expecting for this?
```
ninetynine/git-admin https://github.com/ninetynine/git-admin
ninetynine/react-dynamic-form-select https://github.com/ninetynine/react-dynamic-form-select
// ...
```

Some terminals support clickable links, so only the name could be shown and as a fallback print the example above?

Also- pagination is a thing. So the command would end up being something such as (default `--page` would be `1`):
```
git-admin repo list --page 2
git-admin repo list -p 2
```

cc: @ash123456789 @DivineOmega

**Comment by dextermb** [comment]
Repository list command implemented in 943d515d951da04a38828e894177840cd9f8b553.

Edit: Installable as of `^2.2.0-beta`

**Comment by dextermb** [comment]
Branch protection implemented in 9ea946753b472a8124e4a508e14a892fa5293551.

It's worth noting that the API for branch protection is still in pre-release. Current implemented through the CLI are: `set` and `remove`, looking into implementing specific option commands. Such as updating, listing and removing status checks in the future.

Installable as of `^2.6.0-beta`.
